
\setchapterpreamble[ur][.6\textwidth]{%
\dictum[Umberto Eco, \textit{The Name of the Rose} (1954)]{%
``Then why do you want to know?''\\
``Because learning does not consist only of knowing what we must or we can do, but also of knowing what we could do and perhaps should not do.''
}\vskip1em
} 


\chapter[Chapter 3 -- Disentangling evolutionary, plastic and demographic processes underlying trait dynamics: A review of four frameworks]{Disentangling evolutionary, plastic and demographic processes underlying trait dynamics: A review of four frameworks}
\chaptermark{Evolution or plasticity?}
\label{chap:decpop}

Koen J. van Benthem*, Marjolein Bruijning*, \textbf{Timoth\'ee Bonnet}*, Eelke Jongejans$^\dagger$,
Erik Postma$^\dagger$, Arpat Ozgul$^\dagger$ Accepted in Methods in Ecology and Evolution
\begin{itemize}
\item[*] co-first authors
\item[$\dagger$] co-last authors
\end{itemize}

\section{abstract}
\noindent\begin{enumerate}
\item Biologists are increasingly interested in decomposing trait dynamics into underlying processes, such as evolution, plasticity and demography. Four important frameworks that allow for such a decomposition are the quantitative genetic animal model (AM), the `Geber' method (GM), the age-structured Price equation (APE), and the integral projection model (IPM). However, as these frameworks have largely been developed independently, they differ in the assumptions they make, the data they require, as well as their outcomes and interpretation. 
\item Here we evaluate how each framework decomposes trait dynamics into underlying processes. To do so, we apply them to simulated data for a hypothetical animal population. Individual body size was affected by, among others, genes, maternal effects and food intake. We simulated scenarios with and without selection on body size, and with high and low heritability. 
\item The APE and IPM provided similar results, as did the AM and GM, with important differences between the former and the latter. All frameworks detected positive contributions of selection in the high but not in the low selection scenarios. However, only the AM and GM distinguished between the high and low heritability scenarios. Furthermore, the AM and GM revealed a high contribution of plasticity. The APE and IPM attributed most of the change in body size to ontogenetic growth and inheritance, where the latter captures the combined effects of plasticity, maternal effects and heritability. We show how these apparent discrepancies are mostly due to differences in aims and definitions. For example, the APE and IPM capture selection, whereas the AM and GM focus on the response to selection. Furthermore, the frameworks differ in the processes that are ascribed to plasticity and in how they take into account demography.
\item We conclude that no single framework provides the `true' contributions of evolution, plasticity and demography. Instead, different research questions require different frameworks. A thorough understanding of the different definitions of their components is necessary for selecting the most appropriate framework for the question at hand, and for making biologically meaningful inferences. This work thus supports both future analysis as well as the careful interpretation of existing work.

\end{enumerate}

\section{Introduction}

Understanding trait and population dynamics and how the two are intertwined is crucial for predicting population resilience and viability \parencite[e.g.][]{Merila2014}. Hence, which processes shape population-level trait dynamics (i.e. changes in trait distributions over time) is a fundamental question in ecology and evolution, and one which is gaining in urgency given mounting concern regarding the consequences of anthropogenic environmental change for natural populations \parencite[e.g.][]{parmesan2006}.

Phenotypic trait distributions may be altered across generations by genetic (i.e. evolutionary) processes, as well as by non-genetic processes, such as phenotypic plasticity. Since the realisation that evolutionary and ecological processes may act on the same time scale, distinguishing between the role of evolution and plasticity has been the subject of a substantial body of research \parencite{Hairston2005,Gienapp2008,Post2009}. To complicate matters further, changes in the demographic structure of a population may additionally shape trait distributions \parencite{Coulson2008}. Hence, understanding and predicting trait dynamics ideally requires simultaneously taking into account all three processes \parencite{Pelletier2007,Schoener2011}. 

To date, four major frameworks aiming at distinguishing between the role of evolution, phenotypic plasticity and demography have been developed: 1) The quantitative genetic framework, particularly the animal model \parencite[AM; e.g.][]{Henderson1950}, 2) the `Geber method' \parencite[GM;][]{Hairston2005}, 3) the age-structured Price equation \parencite[APE;][]{Coulson2008}, and 4) the application of the APE in conjunction with an integral projection model \parencite[IPM;][]{easterling2000,Ellner2006,Coulson2010}. Several studies have tried to explicitly estimate the relative importance of evolution, plasticity and/or demography using one of these approaches \parencite[e.g.][]{Reale2003, Ezard2009, Ozgul2009, Rebke2010, Becks2012, Morrissey2012sts}. Nevertheless, fully disentangling and quantifying evolutionary, ecological and demographic processes and ultimately predicting the consequential trait dynamics has proven to be problematic \parencite{Gienapp2008,Schoener2011,Merila2014}. At least some of these difficulties can be attributed to the large amounts of (individual-based) long-term data required, which are often unavailable for natural populations \parencite{Clutton-brock2010}. However, even if sufficient data are available, synthesis of the results from the four frameworks is hampered by the fact that they have been developed largely independently of each other. As a consequence, they differ in their focus and aims, and as we show here, they define biological processes in non-equivalent ways.  

Here we provide an overview of the differences, similarities and complementarity of each of these four decomposition frameworks by applying them to the same simulated datasets and comparing their outcomes. Thereby, we evaluate how they quantify the role of different ecological and evolutionary mechanisms in shaping trait dynamics under a range of biological scenarios. Together with a critical review of the theory underlying each of the frameworks, we provide comprehensive insight into their underlying assumptions, as well as the conceptual differences and similarities. This provides a much needed overview of the suitability of each framework with respect to research questions and data availability.


\section{Applying the four frameworks}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{Data simulation}

Although it comes with the loss of some biological realism, using simulated rather than empirical data enables us to evaluate the frameworks under different scenarios and allows for replication. Furthermore, simulated data do not suffer from the complications introduced by missing data. Finally, it provides a reference that aids the comparison between the results of each framework. Importantly, it is not possible to calculate ``true'' contributions of for example evolution without first adopting one of the frameworks and their corresponding definitions, therefore, our simulations allow only for a qualitative assessment. 

Data were simulated using a two-sex individual-based model of a closed population of a hypothetical animal species, implemented in R \parencite{R2014}. Here, we provide a brief overview, while a more complete description can be found in supporting information S1. We also provide the R code on \newline \verb|https://github.com/koenvanbenthem/Disentangling_Dynamics_IBM|. We simulated a single trait, body size $z$. Size at birth is determined by an individual's genotype (10 loci, with 10 alleles each and mendelian inheritance, more details in S1.1), the body size of its mother (i.e. a maternal effect as in \cite{Falconer1965}), and a stochastic component (drawn from a Gaussian distribution; S1.2). Ontogenetic growth results in an increase of body size with age. Growth rate, the proportional increase in body size,  decreases with age, and is further influenced by per-capita food availability (S1.3). Males were randomly assigned to females, who have a $50\%$ chance of becoming reproductive after one year and whose reproductive probability increases with age. The litter size that a female produces depends on per-capita food availability, a stochastic component, and body size (S1.4). Survival probability first increases with age, but starts decreasing after year five, reflecting senescence, and is further influenced by per-capita food availability and body size. Maximum age is 30 years. Furthermore, a trade-off exists between female reproduction and survival, i.e. reproducing at time $t$ decreases survival probability to time $t+1$ (S1.5). 

We simulated fifty time steps (years). After ten years, total food availability started to decline. Every year the available food is divided over all individuals, with some individuals randomly obtaining more than others. Individual food intake affects survival, growth and (female) reproductive success (S1.6). The first ten years were discarded from further analyses to allow the age structure to stabilize (Fig. \ref{simdata}f) The remaining data spanned 40 years (i.e. approximately 13 generations), which is comparable to the length of some of the field studies these frameworks have been applied to \parencite{Clutton-brock2010}. 

To evaluate the behaviour of the frameworks under different circumstances, we simulated four different scenarios. First, survival and fertility selection on body size was either present ($s_+$) or absent ($s_0$). Under the $s_+$ scenarios, there was a positive effect of body mass on survival and on litter size for mothers. Second, the relative importance of genetic variation in shaping body size, commonly measured as heritability, was either high ($h_+$) or low ($h_-$). This was done by using either of two pre-defined genotype-phenotype maps: one with big and one with small variation in the effects of alleles. Furthermore, to keep the phenotypic variance comparable, we decreased the plastic component in birth size in the $h_+$ scenarios. The parameter values for each of the four scenarios (\sh, \sH, \Sh and \SH) can be found in S1.7. To evaluate the effect of stochasticity, each scenario was replicated 100 times.

Fig. \ref{simdata} provides an illustration of some of the key characteristics of the datasets simulated under each scenario. Despite a substantial amount of stochastic variation across replicates within each scenario, clear differences in trait and population dynamics are apparent. As expected, the $s_+$ scenarios show a positive relation between body size and annual fitness, calculated as the sum of survival and litter size to $t+1$, whereas the $s_0$ scenarios do not (Fig. \ref{simdata}e). Furthermore, the proportion of the phenotypic variance attributable to variance in the simulated genotypic values (i.e. broad-sense heritability) was ca. 0.50 in the $h_+$ and 0.08 in the $h_-$ scenario.
\newpage
\begin{figure}[H]
\includegraphics[width = 0.9\textwidth]{FiguresDecPop/fig1}
\caption{Summary of the observed population and trait dynamics of simulated datasets. (a) Trends in population size, (b) changes in mean body size, (c) mean birth size, (d) and genotypic values for body size, (e) relations between body size and yearly individual fitness (sum of survival and litter size at $t+1$), and (f) changes in mean age. Lines indicate the averages across 100 replicates. Polygons show one standard deviation above and below the average. Red lines indicate $s_0$ scenarios (no viability and fertility selection), blue lines indicate $s_+$scenarios (strong viability and fertility selection). Solid lines indicate $h_{-}$ scenarios (low heritability), dotted lines indicate $h_{+}$ scenarios (high heritability). In a-d and f, the white polygon indicates the first 10 years, which are excluded from further analysis. In (e), lines are averaged predictions based on generalized additive models over all replicates.}
\label{simdata}
\end{figure}

Although in all scenarios population size first increased (until year 20) and then decreased (Fig. \ref{simdata}a), the  population size averaged across replicates reached up to 322 and 334 individuals in scenarios \Sh and \SH, whereas in \sh and \sH the maximum average population size was 245 and 252 individuals, respectively. Mean body size first increased rapidly, but decreased in all scenarios between the eleventh and fiftieth year (Fig. \ref{simdata}b): in \sh with (mean $\pm$ SE) \Range{-0.47}{-1.45}{0.63 \; 95\% \; \text{interval}}{0.058}, in \sH with \Range{-0.46}{-1.59}{0.0.68}{0.061}, in \Sh with \Range{-0.75}{-1.87}{0.08}{0.051}, and in \SH with \Range{-0.16}{-1.12}{0.83}{0.057}. Note that the $95\%$ intervals, here and in the rest of the manuscript, are ranges of point estimates across replicates. They reflect the stochasticity of the simulations rather than the precision of the estimates. The standard errors for each average were calculated by dividing the standard deviation of the values of the replicates by 10 (the square root of the number of replicates). A full power analysis of the methods is beyond the scope of this manuscript.

Contrary to average body size, genotypic values for birth size continued to increase only in scenario \SH. Here, the change in average genotypic value (across the entire population) between year 11 and year 50 was \Range{0.62}{0.23}{1.04}{0.022} (Fig. \ref{simdata}d). In \Sh a smaller increase was observed \Range{0.08}{-0.074}{0.24}{0.0083}, whereas \sh and \sH show on average no change in genotypic values. Correspondingly, average birth size increased only in the \SH scenario, with \Range{0.58}{0.092}{1.11}{0.027}, between year 11 and year 50 (Fig. \ref{simdata}c). 


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%% OVERVIEW OF THE FRAMEWORKS AND THEIR APPLICATION
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{Decomposing simulated trait dynamics}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Rather than providing an exhaustive overview of all methods allowing for the decomposition of trait dynamics, we have chosen to focus on four, commonly-used, frameworks. The four frameworks have different data requirements and do not yield identical results. This is illustrated in the following section, in which we analyse the simulated data using each framework.

\paragraph{Animal Model} 
%%% GENERAL ABOUT THE AM
The animal model (AM) is a quantitative genetic method that was developed for commercial breeding \parencite{Henderson1950,Henderson1976}, where it has been used successfully for several decades \parencite[e.g.][]{Lynch1998}. Only recently has it been applied to wild animal \parencite[e.g.][]{Reale2003,Postma2014} and plant \parencite{Stinchcombe2014} populations. For extensive explanations of the AM as applied to natural populations, see \cite{Kruuk2004} and \cite{Wilson2009}. 

The AM is a linear mixed effects model that is fitted to individual-level data and assumes a quantitative genetic model, where a phenotypic trait ($z$) is influenced by a large number of genes with small effects \parencite{Roff2007}. The variance in $z$ is partitioned into genetic and non-genetic sources of variation. Under the assumption that this partitioning is additive (i.e. in the absence of genotype-environment correlations and interactions), $z$ can be written as the sum of a population mean ($\mu$), an additive genetic effect (the breeding value, $a$) and a residual (environmental) value capturing plasticity ($e$), thus $z=\mu+a+e$. Information on the relatedness between individuals (estimated from a pedigree or genetic markers) is used as a constraint in the fit, allowing for the estimation of $a$. If the data allow for it, other components contributing to variation in $z$, such as maternal, common, and permanent environmental effects can be accounted for explicitly. This variance decomposition can be used to estimate genetic change over time\textemdash resulting from, for example, selection or genetic drift. 

There are several ways to estimate evolution within the AM framework (see discussion), but here we illustrate only one. We fitted a univariate AM and quantified the change in the best linear unbiased predictors (BLUPs) for the breeding values over time \parencite[][]{Postma2006,Hadfield2010b}. We used body size as the sole response variable, and intercepts for breeding values, maternal effects, permanent environment, and year were included as random effects. Maternal and permanent environment effects were modelled by fitting maternal and individual identity, respectively. An alternative specification of the maternal effects, more in line with the simulation process, is briefly discussed further below. Age was included as a continuous fixed effect (both as linear and quadratic terms). All fits were performed using the R-package \texttt{MCMCglmm} \parencite{Hadfield2010a} using inverse-Wishart priors with variance and degree of belief both set to 1. The posterior distributions were estimated based on 1,000 MCMC samples, from 50,000 iterations with a thinning interval of 40 and a burn-in of 10,000, thus ensuring that the correlation between successive samples of all parameters is below 10\%. 

We estimated the temporal trend in the BLUPs for all random effects. We accounted for their uncertainty following \cite{Hadfield2010b} by performing a regression of the BLUPs on time for each MCMC sample of the model. This provided a posterior distribution of linear slope coefficients, estimating the change in additive genetic, maternal, and permanent environment effects per time step. More details on the fitted models are given in S2.1.

As depicted in Fig. \ref{frameworkresults}a, in all scenarios the contributions of evolution and individual plasticity were largest, while the contributions of permanent environment and maternal effects were very small. On average, the per year change in breeding values was positive in both scenario \Sh (\Range{0.0013}{-0.0038}{0.0095}{0.0003}) and scenario \SH (\Range{0.014}{0.00021}{0.029}{0.0007}). Note that the large error bars in Fig. \ref{frameworkresults}a) mostly reflect a substantial amount of variation in the rate of evolutionary change among replicates due to genetic drift, rather than the uncertainty in the point estimates. Negative contributions of individual plasticity were found, particularly in the scenarios with selection \Range{-0.02}{-0.049}{0.0018}{0.0013} and \Range{-0.019}{-0.045}{0.0029}{0.0013} for $h_{-}$ and $h_+$, respectively.

Despite substantial drift, we would expect the contribution of evolution averaged over replicates to be 0 in the $s_0$ scenarios. Instead, our model inferred a genetic decline for $h_{-}$ and $h_+$  of \Range{-0.0057}{-0.016}{0.0040}{0.0005} and \Range{-0.0073}{-0.024}{0.0087}{0.0009}, respectively. The AM therefore estimates evolution with a negative bias. The reason is a mismatch between the model structure and the simulation process. As mean size decreases with time, the maternal contributions to birth size decreases. Because we modelled maternal effects as maternal identity rather than maternal current size, this change is mistaken for evolution. We performed an additional analysis using maternal size instead of maternal identity, which strongly reduced this artefact (details and results in S2.2).

\paragraph{Geber method}
%%% GENERAL DESCRIPTION GM
The `Geber method' (GM) \parencite{Hairston2005} is a very general method that quantifies how temporal changes in various factors influence the response variable of interest. Because of this generality, the biological assumptions depend on the specific implementation. The GM may for example estimate how temporal changes in mean breeding value $\overline{a}$ and in an environmental factor $k$ such as food availability propagate to a population-level response variable $X$, such as mean trait value. Examples of its application can be found in \cite{Ellner2011} and \cite{Becks2012}.

%%% OUR APPLICATION OF THE GM
Our implementation of the GM follows the analysis of fledgling mass in \cite{Ellner2011}. We took the average body size ($\overline z$) as the population-level response variable, and decomposed the change in $\overline z$ into a contribution of the environment ($\overline k$) and a contribution of a phenotypic change in size at birth. The latter was decomposed further into an evolutionary ($\overline a$) and a plastic component ($\overline p$):

\begin{equation}
\frac{d \overline{z}}{dt} = \frac{\partial \overline{z}}{\partial \overline{k}} \frac{d\overline{k}}{dt} + \frac{\partial \overline{z}}{\partial \overline{a}}\frac{d \overline{a}}{dt} + \frac{\partial \overline{z}}{\partial \overline{p}}\frac{d\overline{p}}{dt} 
\end{equation}

For each year between years 11 and 50, we calculated the mean body size ($\overline z$), mean size at birth of newborns, the average food availability that alive individuals had access to during their life up to that moment ($\overline k$), and the mean breeding value as estimated by the AM ($\overline a$) (see above). As breeding values can not be observed directly, the application of the GM to empirical data relies on other methods such as the AM for their estimation. Finally, we calculated a plasticity term ($\overline p$), equal to the difference between the average size at birth and the average breeding value for size at birth. Thereby this term only captured plasticity in mass at birth. 
We fitted a linear model to estimate the effects of $\overline a$, $\overline p$ and $\overline k$ on $\overline z$. Using this model, together with separate linear models that describe how each of the three underlying factors changes over time, we evaluated their respective influence on $\overline z$. This procedure is described in more detail in S3.1.

The results of the GM are shown in Fig. \ref{frameworkresults}b. The results for the evolutionary component are, as expected, nearly identical to the results of the AM. This evolutionary component is counter-acted by a decrease in food availability, as is shown by the negative `environmental' contributions. The latter is largest for the $s_+$ scenarios, under which population size is higher (Fig. \ref{simdata}a) and per capita food availability therefore lower. 

The average contributions of plasticity are more equivocal. Whereas we expected the slight reduction in maternal body size, and hence in the maternal effect, to result in a minor negative contribution of plasticity, we instead see mainly positive contributions. This is the result of the downwardly biased trend in the breeding values (as discussed above). When the analysis was repeated with the `true' genotypic values from the simulations instead of the estimated breeding values, all scenarios showed negative contributions of plasticity (S3.2).

\newpage
\begin{figure}[H]
\includegraphics[width = 0.9\textwidth]{FiguresDecPop/fig2}
\caption{Results of the different frameworks when applied to the simulated scenarios. (a) Animal model. (b) "Geber" method. (c) Age-structured Price equation and (d) Integral projection model. In (c) and (d), demography includes changes in average body size due to the age structure, inheritance is the sum of offspring mother difference and offspring difference covariance. In (a-d), red bars indicate $s_0$ scenarios, blue bars indicate $s_+$ scenarios. Solid bars indicate $h_{-}$ scenarios, and shaded bars indicate $h_{+}$ scenarios. Error bars represent the range in which 68\% (error bars until horizontal lines) and 95\% (entire error bars) of the contributions lie when applied to 100 replicates. The y-axis is always average contribution to mean trait change per year, although the scaling is different in (a), (b) versus (c), (d).}
\label{frameworkresults}
\end{figure}
\paragraph{Age-structured Price Equation}
%%% GENERAL DESCRIPTION APE
The age-structured Price equation (APE) \parencite{Coulson2008} is an extension of the Price equation \parencite{Price1970}. The APE does not explicitly consider genetic variation. It decomposes the change in mean trait value into seven additive components. All these contributions are either averages of, or covariances between, observable individual properties (e.g. individual survival and body size). 

The two selection terms describe how selective disappearance (viability selection, VS) and selective reproduction (fertility selection, FS) alter the mean trait value. Here, VS is the covariance between $z$ and survival, which scales with the difference in the average trait value of the whole population and the part of the population that survives to the next time step \parencite[e.g.][]{rebke2012}. This is referred to as the selection differential in the evolutionary literature \parencite{Robertson1966,Lande1983}. The contribution to the change in mean trait value due to ontogenetic development of surviving individuals is captured by the growth term. The two inheritance-related contributions were combined into one (S4.3). This combined term measures the contribution to changes in average body size due to the difference between the mother's body size (at time of giving birth) and her offspring's body size at birth (i.e. between generations). Because offspring are generally smaller than mothers, the inheritance contribution will typically be negative. This stresses that the inheritance term should not be confused with heritability, which can not be negative.
Finally, the two demography contributions, here also combined into one, describe change resulting from the age structure (S4.2). The demography term arises because the other contributions are calculated per age class. This takes into account that their values depend not only on the trait value of an individual, but also on its age. The total contribution is obtained by a weighted sum of the age specific contributions. 

The APE thus allows for an exact decomposition of $\Delta \overline z$ in discrete time into components of viability selection, fertility selection, ontogenetic growth, inheritance, and demography in populations with overlapping generations.  It has been applied to a range of mammals species \parencite{Coulson2008,Ozgul2009,Ozgul2010,Canale2016}. See S4.1 for the full equation and an explanation of the terms. Note that a  stage-structured version of the Price equation has also been developed \parencite{Barfield2011}.

%%% OUR APPLICATION OF THE APE AND RESULTS
As is commonly done in demographic analyses, we applied the APE to the female part of the population only. Under the $s_0$ scenarios, we find that the average VS and FS are both indistinguishable from zero (Fig. \ref{frameworkresults}c). For the $s_+$ scenarios, the contribution of selection is positive, and there is no difference between the $s_+ h_+$ and $s_+h_-$ scenarios (VS: \Range{0.081}{0.060}{0.10}{0.0012} and \Range{0.090}{0.063}{0.11}{0.0013} respectively, FS: \Range{0.054}{0.027}{0.079}{0.0015} and \Range{0.055}{0.029}{0.082}{0.0015} respectively). Finally, the demographic contribution differs between the $s_0$ and $s_+$ scenarios, but does not differ between $h_+$ and $h_-$. This combined demography term scales with the between-age class covariance between fitness and body size (S4.2). In agreement with our simulation processes, this covariance is strong and positive, as older age classes have larger average body size, and larger individuals have higher fitness in the $s_+$ scenarios. The negative contribution in the $s_0$ scenarios is the result of a negative effect of age on survival, which in the absence of positive selection will dominate the between-age class covariance. The biggest contribution to changes in average body size comes from ontogenetic growth. This component is slightly lower in the $s_+$ scenarios, due to smaller per capita food availability.

The inheritance term is more negative in the $s_+$ than in the $s_0$ scenarios. This is because in the $s_+$ scenarios larger mothers produce more offspring, which on average results in a larger difference between mother and offspring size: although the maternal trait value when giving birth is higher, their offspring's trait value at birth does not increase by the same amount. This leads to the average contribution of inheritance becoming more negative. Furthermore, we see that contributions from inheritance are slightly smaller (less negative) under the $h_+$ scenarios than under the $h_-$ scenarios. This is because with increasing heritability, the mother-offspring difference decreases, leading to a less negative inheritance term.

\paragraph{Integral Projection Model}
%%% GENERAL DESCRIPTION OF IPM
The integral projection model (IPM) is a general model for projecting continuous distributions in discrete time. When describing a population, it often considers four life history processes: survival, reproduction, growth and inheritance \parencite{Ellner2006}. The dependencies of these processes on a continuous phenotypic trait $z$ are estimated using regression models. No assumptions concerning the underlying genetics are made. Based on these regressions, the trait distribution at time $t+1$ can be predicted from the trait distribution at time $t$ \parencite[as well as demographic properties, such as population growth rates, e.g.][]{Adler2010,merow2014}.  Over the past years, IPMs have been used to address a range of eco-evolutionary questions \parencite[e.g.][]{metcalf2008evolution,Smallegange2013,traill2014demography}. While the specific decomposition we use involves applying the APE to a fitted IPM, as proposed by \cite{Coulson2010}, approaches using a sensitivity analysis also exist \parencite[e.g.][]{Coulson2011modeling,traill2014demography}.

%%% OUR APPLICATION
An IPM was parametrized for each simulated dataset, and as we did for the APE, we only considered females. Models describing individual growth, survival and reproduction (both the probability of reproducing and the number of offspring) were fitted using generalized linear mixed models with appropriate link functions (logit for survival and reproduction probability, log for number of offspring). The contribution of inheritance was estimated as a linear regression of offspring size at birth on the size of the mother at the time of giving birth, as done in \cite{traill2014demography}. This differs fundamentally from heritability ($h^2$), where offspring size is related to the mother's size, both at the same fixed developmental stage (e.g. birth) \parencite{Chevin2015a}. For all life history processes, we tested five different models: a full model containing age, size and their interaction, as well as all models nested within this full model. Furthermore, each model included a random effect for year. The model with the lowest AIC was selected and used for the IPM.

Using the selected models, a $3100\times3100$ matrix was parametrized (i.e. 31 age classes, 100 size classes per age class, ranging between 1 and 50) for each replicate. See S5 for more details on model fitting and the construction of the IPMs. For each IPM, we used the observed population vector at each time step (excluding the first ten years) to project the population vector to the next time step ($t+1$). Changes in population structure, and thereby changes in $\overline z$, are decomposed into contributions from different life history processes. 

%%% OUR RESULTS
We found very similar patterns as in the APE (Fig. \ref{frameworkresults}d). Both viability and fertility selection were detected in the $s_+$ scenarios (VS was \Range{0.045}{0.026}{0.063}{0.00096} and \Range{0.041}{0.024}{0.060}{0.00098}; FS was \Range{0.012}{0.00}{0.026}{0.00074} and \Range{0.012}{-0.0044}{0.028}{0.00076}, for $h_{-}$ and $h_+$). In contrast, in the $s_0 h_-$ and $s_0 h_+$ scenarios, average viability selection was \Range{-0.024}{-0.045}{-0.0024}{0.0011} and \Range{-0.019}{-0.039}{-0.00024}{0.0010}, respectively, and fertility selection was \Range{-0.00069}{-0.014}{0.012}{0.00069} and \Range{0.00068}{-0.011}{0.014}{0.00059}.  As in the APE, the contribution of inheritance to $\Delta \overline z$ was large and negative in all scenarios, and was more negative in the $s_+$ scenarios. Furthermore, there was a consistently positive contribution of ontogenetic growth, with weaker effects in the $s_+$ scenarios, again due to lower per capita food availability. As in the APE, we considered both demographic terms together. This term showed positive contributions in all scenarios.

To allow for a better comparison with the other three frameworks, here we focus on the average value of $\Delta \overline z$, and how much various processes contribute to this. When quantifying how much of the \emph{year-to-year variation} in $\Delta \overline z$ is explained by each process \parencite[as for example in][]{Ozgul2009}, the IPM and APE provide more divergent results (S6).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%% Discussion
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Discussion}
We have decomposed changes in mean body size into underlying processes by applying four major  frameworks to simulated data. Thereby we have shown that these frameworks differ substantially in their data requirements, which processes they consider, how these are defined, and how changes in the mean trait value are assigned to them. In the following sections we will discuss and compare the theory underlying the four frameworks, illustrated by our simulations. We will discuss the inherent differences among frameworks regarding evolution, plasticity, demography, and measures of uncertainty. These are summarised in Table \ref{table:questions}. We finish by discussing each framework with respect to data availability and the research question at hand.


\begin{table}[ht] 
\caption{A selection of research questions and to what extent frameworks may be used to answer them, ranging from impossible without major modifications ($--$) to being answered by the standard formulation of the framework already ($++$). AM = animal model, GM = Geber method, APE = age-structured Price equation and IPM = integral projection model. Note that scores are based on the specific application of the frameworks as we reviewed here; this involves the univariate AM, and the application of the APE to the IPM, in case of the IPM. Alternative approaches of the frameworks are mentioned in the discussion.} 
\label{table:questions}
\begin{center}
\begin{tabular}{p{9cm}|p{1cm} p{1cm} p{1cm} p{1cm}}
\hline
Question & AM & GM & APE & IPM\\
\hline
Does the change in trait value have a genetic basis? & $++$ & $+$ & $--$ & $--$ \\

Is selection acting on the trait? & $+$ & $+$ & $++$ & $++$ \\

Is the trait heritable? & $++$ & $\pm$ & $-$ & $-$ \\

Is the age structure responsible for the change in mean trait value? & $+$ & $\pm$ & $++$ & $++$ \\

How does individual heterogeneity affect trait value $z$? & $+$ & $\pm$ & $--$ & $-$ \\

How do trait dynamics affect population dynamics? & $-$ & $+$ & $-$ & $++$ \\

Is an environmental change responsible for the change in mean trait value? & $+$ & $++$ & $--$ & $-$\\

\end{tabular}
\end{center}
\end{table}

We have simulated scenarios with and without selection on body size, and with low and high heritability. As multiple processes influence and interact with body size, these scenarios resulted in divergent and relatively complex population and trait dynamics (Fig. \ref{simdata}). For example, in addition to genetic effects, size at birth was influenced by maternal effects and stochasticity. Moreover, ontogenetic growth was subject to both stochastic variation and a decrease in per-capita food availability. We also included a trade-off between viability and fertility. It is exactly this complexity that highlights the need for a robust framework that allows disentangling the underlying processes and quantifying their importance.

\subsection*{Selection and evolution}
All four frameworks infer positive selection on body size in the $s_+$ scenarios, but not in the $s_0$ scenarios (Fig. \ref{frameworkresults}). The APE and IPM detect positive viability and fertility selection in both the \SH and the \Sh scenarios. The AM and GM detect a strong increase in mean breeding values in the \SH scenario and a small yet positive contribution in the \Sh scenario. Importantly, the AM and GM estimate a genetic change (due to selection and/or drift) whereas the IPM and GPE estimate selection. This is highlighted by the fact that the AM and GM estimate a much larger contribution of evolution in the \SH compared to the \Sh scenario. This contrasts with the IPM and APE, where the contribution of selection is independent of the heritability. 

Due to a misspecification of the maternal effects in the AM, we find a negative contribution of evolution in the $s_0$ scenarios. This mismatch highlights the need to adapt the model structure to the study system. Only then reliable conclusions can be drawn from the AM \parencite[see also][]{Hadfield2011}. Indeed, we show that contributions are closer to the simulation process when we use a more appropriate specification of the maternal effects (S2.2). 

Here we have chosen to quantify the contribution of evolutionary change to trait dynamics by measuring the temporal change in BLUPs for breeding value in a univariate animal model. Within a quantitative genetic framework, we could also have used the heritability estimated by the AM to apply the breeder's equation and estimate the expected response to selection. This approach has proven its effectiveness under breeding conditions, although nonlinearities in the parent-offspring regression or the trait value-fitness relationship may bias predictions \parencite{Heywood2005}. More serious difficulties arise in natural populations, where the prediction of evolution can be biased when selection acts on genetically correlated traits or when an environmental variable dominates the covariation between traits and fitness \parencite{Rausher1992,Morrissey2010}.

A third approach relies on a bivariate AM that estimates genetic and environmental (co)variances between a trait and a proxy for relative fitness \parencite{Lande1979,Lynch2014}. The additive genetic covariance is of particular interest, as following the Robertson-Price identity it provides a direct estimate of the evolutionary change \parencite{Robertson1966,Price1970,Lynch2014}. Although more data demanding, this approach does not require the assumptions of the breeder's equation to be fulfilled \parencite{Morrissey2012sts}, and avoids potentially biased trends in breeding values \parencite{Postma2006}.

Unlike the AM and GM, which quantify the change in breeding values, the APE and IPM estimate the contribution of selection, irrespective of whether this yields a genetic response. The overall contribution of selection is obtained by summing over all age-specific selection contributions. This is an attempt to remove the between-age covariation between traits and fitness \parencite{engen2014b}, which is instead captured by the demography term. However, the age correction is not continuous, and therefore the choice of age classes determines how this total contribution of demography and selection is partitioned (see S4.4 for an example).

Most studies that have applied the APE or IPM framework to natural vertebrate populations have found a relatively small role for selection in shaping trait dynamics \parencite[e.g.][]{Ozgul2009,traill2014demography}. This is in line with our application, as even in the $s_+$ scenarios, the contribution of the other processes was estimated to be many times larger. In the IPM, the interpretation of selection in terms of evolutionary potential critically depends on the heritability. Heritability is, however, not assessed by the IPM. Indeed, the inheritance function relates juvenile to adult (maternal) trait values, and ignores the fact that individual growth trajectories may be heritable \parencite{Chevin2015a}. Alternatively, trait inheritance can be incorporated in the IPM by implementing size at birth as a fixed trait influencing offspring size \parencite{vindenes2015}, or by explicitly modelling the transmission of additive genetic effects within the IPM \parencite{Coulson2015,Childs2016}.

\subsection*{Plasticity}
Plasticity includes all individual-level phenotypic changes that are not attributable to genetic changes. While all four frameworks estimate a large contribution of plasticity in all scenarios, they attribute them to different biological processes. This makes it difficult to directly compare the importance of plasticity across frameworks and may potentially lead to confusion. In this section we will focus on plasticity in birth size.

We used the AM to separately estimate plasticity due to maternal and permanent environment effects (Fig. \ref{frameworkresults}a). The contribution of maternal effects was very small. This may seem at odds with the effect of maternal adult size on offspring size at birth in our simulations, but as explained above, this was due to a mismatch between the model structure (which included a random effect of maternal identity) and the data generating process  (which included an effect of maternal body size). The contribution of permanent environment was low, which is in line with the lack of a trend in the stochastic component of birth size in our simulations.

The GM captures plasticity in size at birth due to both maternal effects and stochasticity in one single term (Fig. \ref{frameworkresults}b). Because plasticity at birth is here defined as the difference between actual birth weight and the breeding value for birth weight of an individual, by construction, the plasticity term has to compensate for the bias in estimated breeding values.

In the APE and IPM frameworks, plasticity at birth and growth are intrinsically entangled. Whereas ontogenetic growth forms the main plastic contribution to $\Delta \overline z$ (Figs. \ref{frameworkresults}c and \ref{frameworkresults}d), the body size that is attained through ontogenetic growth is only partially (through maternal effects) transmitted to the offspring. Most of the ontogenetic growth will thus be reset in the offspring: this is reflected in the strong negative contribution from inheritance (for a more detailed explanation of the inheritance terms, see S4.3.1). Also, because we applied the APE only on the female part of the population, changes in offspring body size due to selection on males (and thus fathers) will be attributed to the inheritance term.

\subsection*{The role of the environment}
Whereas the GM defines an explicit environmental factor, in the other frameworks, the environment influences trait dynamics only indirectly through selection, plasticity and/or demography. For example, high food availability may lead to an increase in average body size through plasticity. At the same time, increased food availability may decrease competition, and thereby affect selection.

In our implementation of the GM, we defined the environment as the total food intake of an individual. Hence, the environment mainly acts through within-individual plasticity through its effect on ontogenetic growth. Importantly, the outcome of the GM depends fully on how evolution, plasticity and environment are defined. When applying the GM to field data, where not all processes are known, it is thus crucial to first identify the main drivers and attribute them to evolutionary, plastic or demographic processes.

Although in the APE and IPM effects of the environment are implicitly present in all terms, in our implementation there is no explicit
quantification of this environmental effect. Although an IPM can include an environmental variable, its contribution will not be quantified by the APE when applied to that IPM. However, alternative applications of the IPM that allow exploring the effects of such an environmental variable do exist \parencite[e.g.][]{vindenes2011}. Alternatively, one can parametrize different IPMs for different environments \parencite[e.g.][]{Ozgul2010} and use comparison methods such as life table response experiments to see how population and trait dynamics differ between these environments \parencite{rees2009}.

In our version of the AM, all contributions of changes in the environment, such as decreasing food availability, are captured within the residual individual plasticity term. Although not commonly done, environmental contributions can be estimated more explicitly by including additional fixed or random effects \parencite{Charmantier2014book}. One possibility is the inclusion of a fixed effect of food availability. Furthermore, it is possible to model interactions between the environmental variable and the additive genetic effects. 

\subsection*{Demography}
We showed how the combined demography terms in the APE scale with the covariance of age class-specific fitness and age class-specific average body size. The demography terms hence do not reflect the effect of changes in the age structure between time $t$ and $t+1$, but rather differences due to the existing age structure at time $t$. As such it provides a demographic correction of estimates of selection, similar to the one proposed by \cite{engen2014b}.

In the AM we have quantified the demographic contribution by multiplying the slope of body size with respect to age with the predicted change in average age. This contribution is most negative in the $s_+$ scenarios, meaning that here a change (decrease) in the average age in the populations over time led to a decrease in the average body size in these scenarios, in agreement with the observed slight decrease in average age as shown in Fig. \ref{simdata}f.

\subsection*{Unexplained variation and uncertainty}
Making conclusive statements regarding which factor has the largest influence on $\Delta \overline z$ requires a measure of the uncertainty in the estimates of each contribution. So far we have only considered the range of point estimates over the replicates, generally showing smaller ranges for APE and IPM. However, APE and IPM were estimating processes that were constant throughout replicates (e.g. selection), whereas the AM and GM were estimating quantities subject to stochasticity (e.g. genetic drift). Differences in range are thus due to the stochasticity in the simulations rather than the uncertainty in the point estimates.

While the AM allows the estimation of confidence intervals for each estimated contribution, in our implementation of the IPM, APE and GM there is no direct measure of uncertainty. For the GM, confidence intervals can be obtained using bootstrapping methods \parencite[as in][]{Ellner2011}. As of yet, the lack of uncertainty quantification is a major drawback of the application of the IPM and APE. However, measures of uncertainty accompanying parameter estimates could be propagated to the decomposition, by using bootstrapping, and in the case of the IPM also by MCMC sampling.

Residual variance is explicitly quantified in the AM. The GM does evaluate the residuals of the underlying regressions, but does not include these in the final results \parencite{Ellner2011}. In contrast, the APE is an exact framework and hence the residual variance is zero. However, it is still subject to sampling variance. Although the IPM uses the APE, it is constructed by fitting statistical models to the data, each with their own residual term. 

The AM can also account explicitly for additional sources of variation, by including the corresponding random effects (for example, we incorporated individual identity as a random effect to account for individual heterogeneity that could not be explained by additive genetic variation). IPMs can also include a random individual effect in the underlying fitted functions. This inclusion accounts for individual heterogeneity when estimating vital rates. However, although this individual heterogeneity should explicitly be propagated to the actual IPM \parencite{vindenes2015}, the IPM is often parametrized with the random effect set to zero. Thereby not all individual heterogeneity is accounted for. Setting the random effect to zero might also bias the prediction because of Jensen's inequality \parencite[e.g.][]{Fox2002}. Individual heterogeneity can be incorporated by defining a ``static trait", in addition to the continuous state variable. This static trait does not change during development, and reflects fixed individual heterogeneity caused by e.g. differences in size at birth, genetics or experienced environment \parencite[e.g.][]{Ellner2006,vindenes2015}. The role of individual heterogeneity is not captured in the GM and APE. In case of the GM, the effects of individual heterogeneity, as estimated by the AM, can be propagated to the response variable.

\subsection*{Conclusions and future directions}
The urge for a better understanding of eco-evolutionary dynamics is reflected in the range of frameworks that have been developed over the last few years aiming at quantifying the underlying processes \parencite{Pelletier2009,Schoener2011}, especially within the light of the consequences of climate change \parencite{Gienapp2008,Lavergne2010}. Yet, a general, predictive framework is lacking, and applications to field data remain scarce. We have shown that the animal model (AM), `Geber' method (GM), age-structured Price equation (APE) and integral projection model (IPM) frameworks differ in generality and data requirements. Importantly, key processes are defined and interpreted differently in the different approaches. We emphasize that one should be careful when applying one of the frameworks and interpreting the outcomes as being the ``true" contributions of different processes. Indeed, we have shown that each framework has its own set of components and definitions.

All four frameworks have only recently been proposed in their current form, and are only starting to be applied to conservation-related questions. In this review we have explored the frameworks and their assumptions and limitations. Our findings are summarized in Table \ref{table:questions}, where we provide an overview of which framework seems most suitable for which research question. The AM enables estimation of quantitative genetic parameters, and genetic change in particular, that cannot be estimated by the other frameworks. However, the AM, and the estimation on quantitative genetic parameters in general, is data demanding and it can be difficult to isolate confounding sources of variation when data sets are small. When individual data on reproduction, survival and growth are available, and one is interested in explicitly quantifying the contribution of within-age class selection, IPM and APE are logical choices. The AM can explicitly evaluate the effect of individual heterogeneity. Although the IPM can take this information into account as well by fitting mixed effects models, it does not evaluate its effect on trait dynamics. In contrast to the other frameworks, only the GM focuses on population-level parameters, but knowledge (or assumptions) on processes is required beforehand, i.e. it must be known what processes are shaped by evolution (or plasticity) and which by the environment. 

We conclude that in isolation none of the frameworks provides a full picture. Instead, each framework answers different questions and has different data requirements. By highlighting both the similarities and the differences, we hope to have aided in the interpretation of existing work. Furthermore, we hope this work will help researchers interested in eco-evolutionary questions in making an informed choice regarding the most suitable framework for their particular question.

\subsection*{Acknowledgements}
We are grateful to Luis-Miguel Chevin for providing us with constructive feedback throughout the process. We also thank Robert O'Hara, Jarrod Hadfield, Loeske Kruuk and one anonymous reviewer for their help in considerably improving this manuscript. Finally, we thank Tim Coulson for comments on an earlier version of the manuscript.  This work was funded by the Swiss National Science Foundation project grants (\verb|31003A_141110| and \verb|31003A_159462/1| to EP, \verb|31003A_146445| to AO) and an ERC starting grant (\verb|#337785| to AO).

\subsection*{Data Accessibility}
The code for simulating the data that we used is provided as online supplementary information, and can also be found on  \verb|https://github.com/koenvanbenthem/Disentangling_Dynamics_IBM| . 







\printbibliography[heading=subbibliography]
