\setchapterpreamble[ur][.6\textwidth]{%
\dictum[Fyodor Dostoyevsky, \textit{The Idiot} (1868--9)]{%
One can't understand everything at once, we can't begin with perfection all at once! In order to reach perfection one must begin by being ignorant of a great deal. And if we understand things too quickly, perhaps we shan't understand them thoroughly.}\vskip1em

\dictum[dude]{%
things}\vskip1em}

\chapter[Chapter 1: General introduction]{General introduction}
\chaptermark{General introduction}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Variation in fitness}
%variation in evolution
Understanding variation among living things is the heart of evolutionary questioning.


Darwin great discovery was to recognize the variation within species as the fuel generating the astonishing diversity of species themselves.

%fitness
There has been a great deal written about the concept of fitness, and as is common for central scientific concepts REF?, its definition is problematic. Actually, there are many definitions, and sub-definitions. 
In this thesis, what we mean by \emph{fitness} is ...
most adapted to the study framework.
with the difficulty that... 


In this thesis, we will not really deal with the fundamental question of appearance and maintenance of variation in fitness (e.g. lek paradox, see XX), but rather with its proximal sources.

 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Quantitative genetics}

How to measure and make sense of genetic variation?
For over a century, there have been two main approaches. 
These can be summarized as ``bottom-up'' and ``top-down''.
The Mendelians and the biometricians



intronic region of the fat metabolism and food intake-related \emph{lepr} gene

We found a recessive allele (let call the recessive allele \emph{a}, and the dominant allele \emph{A}) associated with lighter individuals (Fig. \ref{fig:leprpheno}\textbf{A}). Homozygotes \emph{aa} were -2.9 g lighter (95\% credibility interval $[0.6;5.1]$), that is, 8\% lighter than the mean. On average, during their lifetime, these \emph{aa} individuals produced one third less offspring than the \emph{AA} individuals (Fig. \ref{fig:leprpheno}\textbf{B}). This strong difference in fitness is however not statistically significant, meaning that it could be the result of chance only.


\begin{figure}[ht]
	\includegraphics[width=0.5\textwidth]{FiguresGeneral/PhenoEffect-1}
	\includegraphics[width=0.5\textwidth]{FiguresGeneral/FitnessEffect-1}
	\caption{Body mass and lifetime reproductive success (LRS) as a function of \emph{lepr} genotypes. 
	(\textbf{A}) Expected body mass of snow voles bearing the three \emph{lepr} genotypes. The expectations and 95\% confidence intervals were predicted from a linear mixed model fitted to the 2311 mass measurement of 532 snow voles. The model accounted for sex, age, date of capture and their two-ways interactions, as well as year of capture and multiple measurements of the same individual.
	(\textbf{B}) Expected LRS of snow voles bearing the three \emph{lepr} genotypes. The expectations and 95\% confidence intervals were predicted from a Poisson generalized linear mixed model fitted to the LRS of 611 snow voles. The model accounted for inbreeding coefficient, year of birth and over-dispersion (using an observation-level random effect). For both panels, the dashed horizontal line projects the expected value of genotype \emph{AA} to ease comparison with \emph{Aa} and \emph{aa}.}
	\label{fig:leprpheno}
\end{figure}

Based such a strong phenotypic effect, \emph{lepr} could be called a major locus. But, how much of the genetic variation does \emph{lepr} explain? Knowing the effect of the three genotypes and the allele frequencies one can compute analytically the additive genetic variances associated with a bi-allelic locus \parencite[][p77]{Fisher1941average,Lynch1998}. Thus, the additive genetic variances associated with \emph{lepr} are $0.052$  for body mass and $0.006$ for LRS. For both traits, \emph{lepr} explains about 1\% of the additive genetic variation as estimated from an animal model.

Could we not simply sequence many more genes to explain a larger proportion of genetic variation? Yes but in practice, in small populations and non-model species, one can rarely explain a large share of the additive genetic variation. 
Furthermore, while it is true that a candidate gene approach provide molecular information that goes much beyond a measure of genetic variance, in a small population its usefulness is likely to be limited by the problem of multiple comparisons. 

The uncertainty in the estimation of the effect of \emph{lepr} on mass translates into a Bayesian p-value of $0.01$. 
If other genetic loci were analysed, 


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{This thesis}

\subsection{Objectives}

\subsection{Churwalden snow voles}

\subsection{Thesis outline}


